### PR TITLE
Added a method to retrieve MRs associated to a commit sha 

### PR DESCRIPTION
**Pull request by Agreiwen**

### CODE REVIEW

File: NGitLab.Tests/Docker/GitLabTestContext.cs [hunk 1]
```diff
@@ -180,6 +185,30 @@ void AddSomeCommits()
                         }));
                 }
             }
+
+            void AddMergeRequest()
```
Comment (meziantou) [inline NGitLab.Tests/Docker/GitLabTestContext.cs]: You can already use `CreateMergeRequest()`. I don't think we need a new method.

File: NGitLab/ICommitClient.cs [hunk 2]
```diff
@@ -21,5 +22,10 @@ public interface ICommitClient
         /// Get a specific commit identified by the commit hash or name of a branch or tag.
         /// </summary>
         Commit GetCommit(string @ref);
+
+        /// <summary>
+        /// Get merge requests associated to a commit
+        /// </summary>
+        IEnumerable<MergeRequest> GetRelatedMergeRequests(Sha1 sha);
```
Comment (meziantou) [inline NGitLab/ICommitClient.cs]: - GitLabCollection allows synchronous and asynchronous enumerations
- Using an object allows adding new parameters later if needed

```suggestion
        GitLabCollection<MergeRequest> GetRelatedMergeRequestsAsync(RelatedMergeRequestsQuery query);
```

**Review by meziantou — CHANGES_REQUESTED** [review]

File: NGitLab.Tests/Docker/GitLabTestContext.cs (hunk 1 above)
Comment (Agreiwen) [reply NGitLab.Tests/Docker/GitLabTestContext.cs]: Done.

File: NGitLab/ICommitClient.cs (hunk 2 above)
Comment (Agreiwen) [reply NGitLab/ICommitClient.cs]: Done.

**Review by meziantou — APPROVED** [review]

File: NGitLab.Mock/Clients/CommitClient.cs [hunk 3]
```diff
@@ -38,5 +42,20 @@ public JobStatus GetJobStatus(string branchName)
         {
             throw new NotImplementedException();
         }
+
```
Comment (Agreiwen) [inline NGitLab.Mock/Clients/CommitClient.cs:45]: Hi @meziantou @louis-z and @Toa741, I would like you to check this method please.

File: NGitLab.Mock/Clients/CommitClient.cs [hunk 4]
```diff
@@ -38,5 +42,20 @@ public JobStatus GetJobStatus(string branchName)
...
+        public GitLabCollectionResponse<Models.MergeRequest> GetRelatedMergeRequestsAsync(RelatedMergeRequestsQuery query)
+        {
+            using (Context.BeginOperationScope())
+            {
+                var project = GetProject(_projectId, ProjectPermission.View);
+
+                var mergeRequests = project.MergeRequests.Where(mr => mr.Commits.SingleOrDefault(
+                        commit => commit.Sha.Equals(query.Sha.ToString(), StringComparison.OrdinalIgnoreCase)) != null);
+
+                var relatedMerqueRequests = mergeRequests?.Select(mr => mr.ToMergeRequestClient());
```
Comment (Toa741) [inline NGitLab.Mock/Clients/CommitClient.cs]: `mergeRequests?` => is the nullable check necessary?

Comment (Agreiwen) [reply NGitLab.Mock/Clients/CommitClient.cs]: Nope, I fixed that :)

File: NGitLab.Mock/Clients/CommitClient.cs (hunk 3 above)
Comment (Toa741) [reply NGitLab.Mock/Clients/CommitClient.cs:45]: Looks ok :)

Comment (Agreiwen) [reply NGitLab.Mock/Clients/CommitClient.cs:45]: Thanks!

**Review by Toa741 — APPROVED** [review]

**Pull request merged**